pkg/expr/expr.md

```diff
@@ -0,0 +1,49 @@
+# Expression Language   (CEXL)
+
+Istio (mixer) configuration uses an expression language to evaluate selectors and input maps.
+The syntax of the expression language is C-like.
+
+An expression maps typed Attribute Values to a Value. 
+
+```
+f(attrValue0, attrValue1, ..., constant0, constat1, ...) --> Value
+```
+
+## Predicates
+Selector expressions are predicates that map attribute values to booleans.
+```
+f(attrValue0, attrValue1, ..., constant0, constat1, ...) --> Boolean
+```
+### Examples of selector expressions
+
+1. select all traffic that is destined for service1 and source user is admin.
+`target.service == "service1.ns1.cluster" && source.user == "admin"`
+2. select all requests that did not result in a error.
+`response.status_code < 400`
+3. select requests which were initially send to host "aaa".
+`request.header["X-FORWARDED-HOST"] == "aaa"`
+4. select all traffic that is destined for namespace ns1 and source user is admin.
+`target.service == "*.ns1.cluster" && source.user == "admin"`
+
+## Mapping expressions
+Mapping expressions map attribute values to a typed value.
+
+### Examples of mapping expressions
+
+1. evaluates to request.size if available, otherwise 200
+`request.size| 200  --> int`
+2. evaluated to request.user if available, otherwise "nobody"
+`request.user | "nobody" --> string`
+
+Expression may use any valid attributes
+
+## Grammar
```

Comment:
somewhere, it would be nice to have a fully-specified language definition for quick reference